roles/provision-keycloak-apb/tasks/addrealmuser-keycloak.yml

```diff
@@ -0,0 +1,127 @@
+---
+- name: Generate KC development username
+  shell: tr -d -c "a-zA-Z" < /dev/urandom | head -c 20
+  register: kcdev_username
+
+- name: Generate KC development password
+  shell: tr -d -c "a-zA-Z0-9" < /dev/urandom | head -c 20
+  register: kcdev_password
+
+- name: "Generate keycloak {{ ADMIN_NAME }} auth token"
+  uri:
+    url: "{{ KEYCLOAK_HOST }}/auth/realms/master/protocol/openid-connect/token"
+    method: POST
+    body: "client_id=admin-cli&username={{ ADMIN_NAME }}&password={{ ADMIN_PASSWORD }}&grant_type=password"
+    validate_certs: no
+  register: keycloak_auth_response
+  until: keycloak_auth_response.status == 200
+  retries: 600
+  delay: 2
+
+- set_fact:
+    KCDEV_USERNAME: '{{ kcdev_username.stdout }}'
+    KCDEV_PASSWORD: '{{ kcdev_password.stdout }}'
+
+- name: "Create {{ KCDEV_USERNAME }} user in {{ KEYCLOAK_REALM }} realm"
+  uri:
+    url: "{{ KEYCLOAK_HOST }}/auth/admin/realms/{{ KEYCLOAK_REALM }}/users"
+    method: POST
+    body: '{"enabled":true,"attributes":{},"username":"{{ KCDEV_USERNAME }}","emailVerified":"","email":"{{ KCDEV_USERNAME }}@email.com","firstName":"Openshift","lastName":"Developer"}'
```

Comment:
Do we need to pass in the email here? If not, I think it would be better not to since the username is generated.

Reply:
I like with email :)

Reply:
Done!